products.php

```diff
@@ -0,0 +1,58 @@
+<?php
+require_once 'src/Models/CategoryModel.php';
+require_once 'src/Services/ProductsDisplayService.php';
+require_once 'src/Models/ProductModel.php';
+require_once 'src/Factory/furnitureDatabaseConnector.php';
+require_once 'src/Entities/ProductEntity.php';
+
+$db = furnitureDatabaseConnector::connect();
+$error_template = '<h1 class="text-5xl mb-2"> Oops, something went wrong </h1>';
+$products = [];
+$category = null;
+
+if (!empty($_GET['id']) && is_numeric($_GET['id'])) {
+    $id = $_GET["id"];
+    $products = ProductModel::getProductsByCategoryId($db, $id);
+}
+
+if (!empty($products)) {
+        $category = CategoryModel::getCategoryTitle($db, $id);
+    }
```

Comment:
It would be better if you did these checks the other way around, that is first check if the category exists and only then go get the products *if* it exists.

That way, in the scenario where there are products pointing to a category that doesn't exist, you wouldn't print both 'Oops there was an error' *and* a bunch of products. The user could see just the error message, in this erroneous circumstance.